api/v2/api.go

```diff
@@ -566,3 +622,22 @@ func (api *API) setupRoutes() error {
 	api.l.Info("Routes initialized")
 	return nil
 }
+
+// HandleQueueError is used to handle queue connectivity issues requiring us to re-connect
+func (api *API) handleQueueError(amqpErr *amqp.Error, cfg *config.TemporalConfig, queueType queue.Queue, publish bool) (*queue.Manager, error) {
```

Comment:
I think `config.TemporalConfig` is wayyyy to broad of a scope to be passing around here - if anything, replace it with just `string` for now

where possible, we want to narrow the scope of what gets passed around so that each function takes as little as it needs to be functional, that way its dependencies are clear. without digging into this code, it's impossible to know whether `handleQueueError` relies on every single field in config, half of them, or (in this case) just one

Reply:
good point!